packages/typography/src/Typography.tsx

```diff
@@ -136,4 +137,145 @@ const Overline: ExtendableBox<{
   return <Box className={cx(sharedStyles, overline, className)} {...rest} />;
 };
 
-export { H1, H2, Subtitle, Body, InlineCode, Disclaimer, Overline };
+const OpenInNewTab = (props: JSX.IntrinsicElements['svg']) => {
+  return (
+    <svg
+      width="16px"
+      height="16px"
+      viewBox="0 0 16 16"
+      xmlns="http://www.w3.org/2000/svg"
+      xmlnsXlink="http://www.w3.org/1999/xlink"
+      {...props}
+    >
+      <title>Open in New Tab</title>
+      <g stroke="none" strokeWidth="1" fill="none" fillRule="evenodd">
+        <path
+          d="M9.29411765,3 L10.7727647,4.47864706 L6.42788235,8.82352941 L7.17647059,9.57211765 L11.5218824,5.22776471 L13,6.70588235 L13,3 L9.29411765,3 Z M4,4.58823529 L4,12 L11.4122941,12 L11.4122941,7.76470588 L10.3529412,7.76470588 L10.3529412,10.9411765 L5.05882353,10.9411765 L5.05882353,5.64705882 L8.23529412,5.64705882 L8.23529412,4.58823529 L4,4.58823529 Z"
+          fill="#007CAD"
+        />
+      </g>
+    </svg>
+  );
+};
+
+const anchorDataProp = createDataProp('anchor-container');
+
+const linkStyles = css`
+  text-decoration: none;
+  color: ${uiColors.blue.base};
+  cursor: pointer;
+
+  &:focus {
+    outline: none;
+  }
+`;
+
+const underline = css`
+  background-repeat: repeat-x;
+  background-position: 0 1em;
+  background-size: 3px 3px;
+
+  ${anchorDataProp.selector}:hover & {
+    background-image: linear-gradient(
+      to right,
+      ${uiColors.gray.light2} 100%,
+      ${uiColors.gray.light2} 0
+    );
+  }
+
+  ${anchorDataProp.selector}:focus & {
+    background-image: linear-gradient(to right, #9dd0e7 100%, #9dd0e7 0);
+  }
+`;
+
+const arrowRightIconPermanent = css`
+  transform: translate3d(3px, 0, 0);
+`;
+
+const arrowRightIconHover = css`
+  height: 10px;
+  width: 10px;
+  opacity: 0;
+  transform: translate3d(-3px, 0, 0);
+  transition: all 100ms ease-in;
+
+  ${anchorDataProp.selector}:hover & {
+    opacity: 1;
+    transform: translate3d(3px, 0, 0);
+  }
+`;
+
+const openInNewTabStyles = css`
+  margin-bottom: -1px;
+  margin-left: -1px;
+  margin-right: -2px;
+`;
+
+const CanShowArrow = {
+  Hover: 'hover',
+  Permanent: 'permanent',
+} as const;
+
+type CanShowArrow = typeof CanShowArrow[keyof typeof CanShowArrow];
+
+export interface LinkProps extends HTMLElementProps<'a'> {
+  canShowArrow?: CanShowArrow;
+}
+
+const Link: ExtendableBox<LinkProps, 'a'> = ({
+  href,
+  children,
+  className,
+  target: targetProp,
+  canShowArrow,
+  ...rest
+}: LinkProps) => {
+  if (!href) {
+    return null;
+  }
+
+  const size = useBaseFontSize();
+  const fontSize = size === 16 ? typeScale2 : typeScale1;
+
+  const hrefHostname = new URL(href).hostname;
+  const currentHostname = window.location.hostname;
```

Comment:
Not server side friendly.  Not sure we have a standard way of being server side friendly yet.

Reply:
Can we keep this logic but either expose a prop or default to an empty string for graceful degradation in the case where we don't have `currentHostname`

Reply:
@DesignerDave thoughts?

Reply:
We have a build variable we can use in a conditional to only use the window when the target is a browser. Check out our emotion package to see how this is done there.